src/extensions/changelog.ts

```diff
@@ -0,0 +1,85 @@
+import * as marked from "marked";
+import * as clc from "cli-color";
+import * as semver from "semver";
+import TerminalRenderer = require("marked-terminal");
+import Table = require("cli-table");
+
+import { listExtensionVersions, parseRef } from "./extensionsApi";
+import { logger } from "../logger";
+import { logLabeledWarning } from "../utils";
+
+marked.setOptions({
+  renderer: new TerminalRenderer(),
+});
+
+/*
+ * getReleaseNotesForUpdate fetches all version between toVersion and fromVersion and returns the relase notes
+ * for those versions if they exist.
+ * @param extensionRef
+ * @param fromVersion the version you are updating from
+ * @param toVersion the version you are upodating to
+ * @returns a Record of version number to releaseNotes for that version
+ */
+export async function getReleaseNotesForUpdate(
+  extensionRef: string,
+  fromVersion: string,
+  toVersion: string
+): Promise<Record<string, string>> {
+  const releaseNotes: Record<string, string> = {};
+  const filter = `id<="${toVersion}" AND id>"${fromVersion}"`;
+  const extensionVersions = await listExtensionVersions(extensionRef, filter);
+  for (const extensionVersion of extensionVersions) {
```

Comment:
This assumes `listExtensionVersions` returns in a sorted manner, which is not part of the contract from our backend
https://source.corp.google.com/piper///depot/google3/firebase/mods/go/spanner/db.go;l=1330;rcl=391185722

So it'd be safer to sort it here so users are guaranteed to see all versions in incremental order